sqllogictest/src/parser.rs

```diff
@@ -679,6 +712,34 @@ mod tests {
             UsefulDiffDisplay(&changeset),
             output_contents,
         );
+
```

Comment:
IMHO if `unparse(parse())` is identity (i.e., the above assert passed), then `parse(unparse(parse()))` isn't necessary. Should we remove the assert above? 🤔

Reply:
🤔 I really like it :) But I can remove it if you prefer

Reply:
I doesn't dislike it, but I proposed `parse(unparse(parse()))` just becasue `parse_roundtrip` can fail in such cases.

```
        // Technically this will not always be the same due to some whitespace normalization
        //
        // query   III
        // select * from foo;
        // ----
        // 1 2
        //
        // Will print out collaposting the spaces between `query`
        //
        // query III
        // select * from foo;
        // ----
        // 1 2
```

If we don't care or test that case, I think we don't need to test `parse(unparse(parse()))` at all. The old one is enough.. 🤪 How do you think?

Reply:
For my usecase downstream https://github.com/apache/arrow-datafusion/issues/4570, just need the `parse(unparse(parse()))` guarantee.

I don't really need the `parse_roundtrip` guarantee. It would be nice if the results of printing a `Record` will look pretty close to its input so that the "test script completion mode" is minimally disruptive to the rest of the records

Thus I think I will try add specific tests for reprinting certain records and change the tests for parsing the example files to ensure that the files survive roundtrip